osu.Game.Rulesets.Taiko/Objects/Drawables/DrawableHit.cs

```diff
@@ -209,6 +209,8 @@ protected override void UpdateHitStateTransforms(ArmedState state)
 
                     this.ScaleTo(0.8f, gravity_time * 2, Easing.OutQuad);
 
+                    MainPiece.X = -X; // Visually snap hit object to hit target
```

Comment:
This just cannot work. When rewinding a replay, hits will be left in this position.

This needs to be done with a `.MoveToX` at very least.

https://user-images.githubusercontent.com/191335/212815729-02ddd875-c6f9-44f6-a70c-4abddb500e17.mp4


Also hitting too early is *very* weird now, but I guess THIS IS WHAT WE WANT?

https://user-images.githubusercontent.com/191335/212815969-37c05f38-5d56-485e-af08-8419a469f1f6.mp4

Reply:
> Also hitting too early is very weird now, but I guess THIS IS WHAT WE WANT?
If you want, we can go back to the version where only great hits were snapped, as it kind of stops this from looking like that

Reply:
Please see my final comment on this PR. This needs a discussion opened, and needs research to be done on what TnT used, what people expect, and what we want.

This is not about subjective opinions. It's about making the correct call going forward. There's no hurry to make a change.

Reply:
👍